backend/routes/shifts.ts

```diff
@@ -45,4 +47,45 @@ router.post('/', async (req: any, res: any) => {
   }
 });
 
+// update a shift in the database
+router.patch('/:id', async (req: any, res: any) => {
+  try {
+    const { id } = req.params;
+    const updates = req.body;
+    // returns the deleted document
+    const oldShift = await Shift.findByIdAndUpdate(id, updates);
+    // get old & new hours for user update later
+    const newTotalHours = updates.hours - oldShift.hours;
```

Comment:
to find new total hours would we want to fetch the user's current total hours and subtract the difference between the oldShift hours and update.hours? unless i'm misreading the code

Reply:
Ahhh, we actually only need the difference between their old total and their new total since we are making use of the `$inc` operator. I should rename `newTotalHours` to `totalHoursDiff` or `hoursIncrement` for clarity.

Reply:
My bad, that variable was only meant to be used to calculate the difference in hours since we use the `$inc` operator to do the update.  
I will rename it to `hoursDiff` for clarity because that is indeed confusing and I even used it by accident in a place where I meant to do the actual new total hours calculation